### PR TITLE
ci: speed up win64 test

### LINKED ISSUE #8696: Windows CI test is extremely long (~3 hours)
### Describe the bug

Some examples:

- https://github.com/apache/arrow-datafusion/actions/runs/7369048334/job/20053938917
- https://github.com/apache/arrow-datafusion/actions/runs/7364543678/job/20045028037
- https://github.com/apache/arrow-datafusion/actions/runs/7358744456/job/20032462383

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

Not sure if it was like this in the past too? But a 3hr runtime seems odd

### PR DESCRIPTION
**Pull request by Jefffrey**
## Which issue does this PR close?

Relates to #8696

## Rationale for this change

Apply changes per analysis from https://github.com/rust-lang/rust/issues/119560

### CODE REVIEW

**Comment by Jefffrey** [comment]
Limit to running only tpcds_physical_q44 with backtrace feature:

good (2023-10-29 nightly) = 70 seconds

bad (2023-10-30 nightly) = 692 seconds

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -310,11 +310,10 @@ jobs:
           cd datafusion-cli
           cargo test --lib --tests --bins --all-features
         env:
-          # do not produce debug symbols to keep memory usage down
-          # use higher optimization level to overcome Windows rust slowness for tpc-ds
-          # and speed builds: https://github.com/apache/arrow-datafusion/issues/8696
-          # Cargo profile docs https://doc.rust-lang.org/cargo/reference/profiles.html?profile-settings#profile-settings
-          RUSTFLAGS: "-C debuginfo=0 -C opt-level=1 -C target-feature=+crt-static -C incremental=false -C codegen-units=256"
```
Comment (Jefffrey) [inline .github/workflows/rust.yml:317]: Removing the other optimization flags as it caused compilation to take 27 min

Without them, it takes 8 min

**Comment by Jefffrey** [comment]
win64 test ran in 33min

Still slightly slower than before (around 27min I believe)

cc @comphead @alamb

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -310,11 +310,10 @@ jobs:
...
+          # Minimize producing debug symbols to keep memory usage down
+          # Set debuginfo=line-tables-only as debuginfo=0 causes immensely slow build
```
Comment (alamb) [inline .github/workflows/rust.yml:314]: 👍  thank you

**Review by alamb — APPROVED** [review]
Thank you very much @Jefffrey This looks really good to me. Also watching you debug this issue I think is a master class in careful engineering. Very impressive 💪 

Screenshot 2024-01-06 at 9 44 33 AM

**Review by comphead — APPROVED** [review]
oh, wow, so debuginfo another level fixed the problem, nicely done @Jefffrey

**Pull request merged**